lib/px/service/client/base.rb

```diff
@@ -4,6 +4,19 @@ class Base
     include Px::Service::Client::CircuitBreaker
     cattr_accessor :logger
 
+    def initialize
+      # monitors circuit state and throws CircuitBrokenException
+      # if state is open. Similar to `handle` function of circuit_handler
+      handler = self.class.circuit_handler
+      Typhoeus.before do
+        if handler.is_tripped(self.circuit_state)
```

Comment:
If the circuit is half-open I think we'll end up letting a lot of requests through (since it doesn't transition back to `open` or `closed` until it knows the success/failure result of the request -- ie, the breaker state object assumes a single request at a time.